autogpt/commands/execute_code.py

```diff
@@ -8,9 +8,39 @@
 
 from autogpt.commands.command import command
 from autogpt.config import Config
+from autogpt.config.ai_config import AIConfig
 from autogpt.logs import logger
 
 
+@command(
+    "execute_python_code", "Execute Python Code", '"code": "<code>", "name": "<name>"'
+)
+def execute_python_code(code: str, name: str, config: Config) -> str:
+    """Execute Python code in a Docker container and returns the STDOUT of the
+    executed code. If there is any data that needs to be captured use a print
+    statement
+
+    Args:
+        code (str): The python code to run
+        name (str): A short description of what the code does
+
+    Returns:
+        str: The STDOUT captured from the code when it ran
+    """
+    ai_name = AIConfig.load(config.ai_settings_file).ai_name
+    directory = os.path.join(config.workspace_path, ai_name, "executed_code")
+    os.makedirs(directory, exist_ok=True)
+    path = os.path.join(directory, name + ".py")
+
+    try:
+        with open(path, "w+", encoding="utf-8") as f:
+            f.write(code)
+
+        return execute_python_file(f.name, config)
```

Comment:
Why not `python -c '<code>'`?

It is not immediately logical for an `execute_python_code` command to create a file (and not clean it up afterwards). If we're going to go that way, we might want to look into a more comprehensive system of generated abilities such as Voyager uses.

Reply:
The reason I chose to use `execute_python_file` is because there's quite a lot of logic about docker containers that I didn't want to copy+paste. The logic in that command is pretty weird and fragile and I didn't want to introduce regressions.

Reply:
Sure, that's fair. In that case, it should be clear from the command description or return message that it creates and executes a file containing the given code.

Reply:
By "command description", I mean the one in the decorator which ends up in the system prompt. ;)

Reply:
🤦‍♂️ fixed